packages/components/src/popover/index.js

```diff
@@ -526,9 +489,12 @@ const Popover = ( {
 				}
 			) }
 			{ ...contentProps }
-			onKeyDown={ maybeClose }
-			{ ...focusOutsideProps }
-			ref={ mergedRefs }
+			ref={ useCallback( mergeRefs( [ containerRef, dialogRef ] ), [
```

Comment:
@ellatrix so you were right about the need of useCallback here :(. My memize approach didn't work for the following reason:

 - memize memorized the result  of the callback function by assuming that React, even if the function instance changes will call the callback using the same node, but this is not the case, when the function ref changes, react first calls the previous function with "null" and the new one with the new node. This was causing the mount to trigger on each render (it's like it unmounted on each render).

So It seems we do need `useMergeRef` but there's one downside the callback refs shouldn't have any dependency. Basically, you can do that and use it as ref:

```
const ref = useCallback( doSoemthingWithNode, deps ); 
```

In this example deps always need to be an empty array otherwise, it won't be called again when the function changes. This is sufficient for our use-cases but I wonder if we can guard against the misusage.

Reply:
Yeah, ref callbacks are quite complex to do right... :( I think we should stick to making it a "real" ref that doesn't change throughout the component lifecycle.

Reply:
I wrote a custom `useMergeRefs` hook in #27768 which would allow you to add dependencies.